tests/Integration/Behaviour/Features/Scenario/Order/generate_invoice.feature

```diff
@@ -0,0 +1,10 @@
+@reset-database-before-feature
+Feature: Generating invoice for Order
+  As a merchant
+  I must be able to generate invoice for Order
+
+  Scenario: Generating invoice for Order that does not have any invoices
+    Given there is order with reference "XKBKNABJK"
+    And order "XKBKNABJK" does not have any invoices
+    When I generate invoice for "XKBKNABJK" order
+    Then order "XKBKNABJK" should have invoice
```

Comment:
Maybe we can assert a few properties from the invoice to check it was correctly generated ?

Also if there is a usecase where an invoice cannot be generated from the order (because of a bad status for example), we can add it here 😄

Reply:
Sure, I have a few questions:
1. Is it ok to use demo data in behat tests?
2. Is this scenario ok? Do you have any more suggestions regarding steps?

Reply:
1. Yes it's OK to use demo data in behat tests if this helps to avoid big setup steps like "I create a Country, I create a Customer, I create an address for this customer for this country, I create a Category, I create a product in this Category ..."

At some point we'll probably have to expand the demo data to have better usecases ready to be used. Like a visitor, a customer with 1 order, a customer with a lot of orders, a customer with a lot of carts...

2. This scenario would benefit from a little more assertions 😉we want to catch bad behaviors created from our modifications of the code. Comparing the invoice amount with the order amount could be interesting for example as it would help us feel safe when modifying the computation method.

Reply:
🆙